tests/storage/restore.go

```diff
@@ -1564,15 +1565,24 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
 					}
 				}
 
-				DescribeTable("should restore a vm that boots from a network cloned datavolumetemplate", func(restoreToNewVM bool) {
-					vm, vmi = createAndStartVM(tests.NewRandomVMWithDataVolumeCloneSourceAndUserData(
+				createVMFromSource := func() *v1.VirtualMachine {
```

Comment:
consider adding variable to either do it with the datavolume template or create the data volume so the vm will boot from the data volume (what you are doing in the second test and there you take the dv info update it create the dv and put nil in the datavolumetemplate